src/puya/awst/nodes.py

```diff
@@ -1224,6 +1226,27 @@ class AppStateDefinition(Node):
     storage_wtype: WType
 
 
+@attrs.frozen(init=False)
+class ScratchSpaceReservation(Node):
+    start_slot: int
+    stop_slot: int
+
+    def __init__(self, start_slot: int, stop_slot: int, source_location: SourceLocation) -> None:
+        if start_slot >= stop_slot:
```

Comment:
Can this be done with an attrs validator? e.g. https://github.com/algorandfoundation/puya/blob/main/src/puya/ir/models.py#L282

Reply:
Um, maybe? But how would that work when we need to check two fields?

Reply:
I've simplified this away so no longer an issue